core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1793,6 +1793,22 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
           OperandTypes.ARRAY_OF_INTEGER,
           SqlFunctionCategory.STRING);
 
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction CODE_POINTS_TO_STRING =
+      SqlBasicFunction.create("CODE_POINTS_TO_STRING",
+          ReturnTypes.VARCHAR_NULLABLE,
+          OperandTypes.ARRAY_OF_INTEGER,
+          SqlFunctionCategory.STRING);
+
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction TO_CODE_POINTS =
+      SqlBasicFunction.create("TO_CODE_POINTS",
+          ReturnTypes.INTEGER
```

Comment:
Just a question not a comment: I think the answer is yes, but is there a reason you can't use INTEGER_NULLABLE and then transform it to array? is it due to the ordering?

Reply:
The order of transform will affect the return types. 
`INTEGER_NULLABLE.andThen(SqlTypeTransforms.TO_ARRAY)` means: an not null array, which element may be null.
`INTEGER.andThen(SqlTypeTransforms.TO_ARRAY).andThen(SqlTypeTransforms.TO_NULLABLE)` means: an array may be null, but its element is not null.

In `to_code_points` function, we can not get a result like`[1, NULL, 2, 3]`, the element type of array never be `NULL`, so i use `INTEGER` instead of `INTEGER_NULLABLE`.

Reply:
Good suggestion! I add this `SqlTypeTransforms`  as you recommended. But a litter different from #3430, because in #3430, `TO_ARRAY_NULLABLE`  is defined as 
`ARG0`
`.andThen(SqlTypeTransforms.TO_ARRAY)`
`.andThen(SqlTypeTransforms.TO_NULLABLE)`, 
which can not be used in my PR.